src/domain-services/flows/model.ts

```diff
@@ -0,0 +1,33 @@
+import { type Database } from '@unocha/hpc-api-core/src/db';
+import { type FlowId } from '@unocha/hpc-api-core/src/db/models/flow';
+import { type Condition } from '@unocha/hpc-api-core/src/db/util/conditions';
+import { type OrderByCond } from '@unocha/hpc-api-core/src/db/util/raw-model';
+import { type InstanceOfModel } from '@unocha/hpc-api-core/src/db/util/types';
+import { type SortOrder } from '../../utils/graphql/pagination';
+import { type EntityDirection } from '../base-types';
+
+export type FlowModel = Database['flow'];
+export type FlowInstance = InstanceOfModel<FlowModel>;
+export type FlowWhere = Condition<FlowInstance>;
+export type FlowFieldsDefinition = FlowModel['_internals']['fields'];
```

Comment:
Getting `FieldDefinition` from `_internals.fields` is incorrect, resulting in `FlowOrderByCond` will have column as `never` type. Can we change this? If it is difficult to get `FieldDefinition` from the model, can we address this in `hpc-api-core`. Due to this, there are a few instances of using `as key of` keyword usage. I don't think this is correct approach.